.circleci/config.yml

```diff
@@ -0,0 +1,162 @@
+# This configuration pipeline is meant to outline structure and naming standards
+# Author: Jonathan Fontaine
+version: 2.1 #CircleCI API Version
+orbs:
+  snyk: snyk/snyk@0.0.8
+
+jobs: #Defines the overall Jobs for this particular pipeline to be used in workflows or as isolated jobs
+  Build: # Execute compilation and library linkage as first step
+    docker: &image # The '&image' setups up a reference tag to be used by '*image' (see unit tests)
+      - image: circleci/openjdk:11-jdk
+    steps:
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - run: &env # does the same as '&image' above and see usage below in unit tests
+          name: Setting environment
+          command: |
+            echo "export DISABLE_DOWNLOAD_PROGRESS_OPTS=-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn" >> $BASH_ENV
+            source $BASH_ENV
+      - checkout
+      - run:
+          name: Maven install without tests
+          command: |
+            mvn install -B -DskipTests $DISABLE_DOWNLOAD_PROGRESS_OPTS
+  Unit Testing: # Execute unit tests separate from other executions to track performance
+    docker: *image
+    steps:
+      - run: *env
+      - checkout
+      - run:
+          name: Maven test phase
+          command: |
+            mvn package test -B $DISABLE_DOWNLOAD_PROGRESS_OPTS
+  Dependency Scan: # Security-Dependency Scanning
+    docker: *image
+    steps:
+      - checkout
+      - snyk/scan:
+          fail-on-issues: true # Fail if we discover any issues/vulnerabilities
+          monitor-on-build: false
+          severity-threshold: high # Set the failure threshold to high
+          token-variable: SNYK_TOKEN #This Token is an environment variable -- API Token from Snyk
+  Container Scan: # Security-Container Scanning
+    docker:
+      - image: circleci/ruby:2.4.1
+    steps:
+      - checkout
+      - run: echo "Scanning Container via Anchore"
+  Docker Image Build: # Build-Container Image -- Call out what flavor (likely always Docker) and indicate this is the Build of said container/image
```

Comment:
Does this build a docker image for the ion-transformation repo and push it a docker registry?

Reply:
I don't think it currently does anything except that print statement. I expect future versions will have options for customizing our Docker builds.

Reply:
I could add the Maven commands in here for doing that though.

Reply:
I wonder how the platform team is planning on handling this.  For example, what docker registry do we push to for master builds?  What docker registry do we push to for PR builds?  Questions like the above make me wonder how all of this is going to work in practice.

Reply:
Actually now that I think about it, shouldn't the Docker images already be created and pushed in the `mvn install` step above in the workflow? It should be pushing to the `r.ion.phx.connexta.com` registry by default.

Reply:
Doesn't one need to run somthing similar to: `mvn clean install docker:build docker:push`?

Reply:
Sorry yeah that's right, my mistake. By default, `mvn install` creates an image in the local registry and when you add `docker:push` it should push to `r.ion.phx.connexta.com`. I don't think PR or local builds should push to our internal registry, but it probably would be good for master builds to do that. We can add that to our CircleCI workflow and have it only run on the master branch.